fendermint/eth/api/src/apis/eth.rs

```diff
@@ -1057,19 +1057,77 @@ pub async fn unsubscribe<C>(
     uninstall_filter(data, Params((filter_id,))).await
 }
 
-use params::{EstimateGasParams, SubscribeParams};
-
-use self::params::TypedTransactionCompat;
+use params::{EstimateGasParams, SubscribeParams, TypedTransactionCompat};
 
 mod params {
     use ethers_core::types::transaction::eip2718::TypedTransaction;
     use ethers_core::types::Eip1559TransactionRequest;
     use ethers_core::types::{self as et, Eip2930TransactionRequest, TransactionRequest};
-    use serde::{Deserialize, Serialize};
+    use serde::{Deserialize, Deserializer};
 
     use crate::state::WebSocketId;
 
-    #[derive(Serialize, Deserialize, Clone, PartialEq, Eq, Debug)]
+    /// Deserialize either the `input` or the `data` field, preferring the former.
+    fn deserialize_input_or_data<'d, D: Deserializer<'d>>(
+        d: D,
+    ) -> Result<Option<et::Bytes>, D::Error> {
+        #[derive(Deserialize)]
+        struct InputOrData {
+            input: Option<et::Bytes>,
+            data: Option<et::Bytes>,
+        }
+
+        let InputOrData { input, data } = InputOrData::deserialize(d)?;
+
+        Ok(input.or(data))
+    }
+
+    /// Parameters for sending a transaction
+    ///
+    /// Copied from ethers because the `go-ethereum` client now sends `input` instead of `data`.
+    ///
+    /// See https://github.com/filecoin-project/lotus/pull/11471
+    #[derive(Clone, Default, Deserialize, PartialEq, Eq, Debug)]
+    pub struct Eip1559TransactionRequestCompat {
+        pub from: Option<et::Address>,
+        pub to: Option<et::NameOrAddress>,
+        pub gas: Option<et::U256>,
+        pub value: Option<et::U256>,
+        #[serde(deserialize_with = "deserialize_input_or_data", flatten)]
+        pub data: Option<et::Bytes>,
+        pub nonce: Option<et::U256>,
+        #[serde(rename = "accessList", default)]
+        pub access_list: et::transaction::eip2930::AccessList,
+
+        #[serde(rename = "maxPriorityFeePerGas", default)]
+        pub max_priority_fee_per_gas: Option<et::U256>,
+
+        #[serde(rename = "maxFeePerGas", default)]
+        pub max_fee_per_gas: Option<et::U256>,
+
+        #[serde(default, rename = "chainId")]
+        /// Chain ID (None for mainnet)
+        pub chain_id: Option<et::U64>,
+    }
```

Comment:
In my local version of this this fix, I had followed a simpler approach of creating Compat structures for all transaction types, and "embedding" the original type in an `orig` field with `#[deref]` and `#[serde(flatten)]`, then only adding the field for input. This solution has us copying the entire type from upstream, which is not bad, but a bit duplicative.

Reply:
Hm, that's a nice idea, thanks, I'll try.

Reply:
Changed it